add new_with_weight

v1xingyue opened the pull request:
Add function new_with_weight to Backend type. That will much more friendly with Weighted RoundRobin .

andrewhavck commented on pingora-load-balancing/src/lib.rs:
```diff
@@ -58,12 +58,17 @@ impl Backend {
+    }
+
+    // Create a new [Background] with address and weight.
+    pub fn new_with_weight(addr: &str,weight: usize) -> Result<Self> {
```
Need to fix the formatting here: `new_with_weight(addr: &str, weight: usize)`

v1xingyue commented:
Ya, that's very clear.

andrewhavck commented:
Thanks for addressing the review comments, however, it looks like there are some merge conflicts.

v1xingyue commented:
> Thanks for addressing the review comments, however, it looks like there are some merge conflicts.

I got it. The main branch add Extension params. I modify that. 
Then still have the conflict:

main branch tell weight always be 1 , my branch tell the weight with value. 
so should accept my changing.

While I don't have permissions to do that.

v1xingyue commented:
> Thanks for addressing the review comments, however, it looks like there are some merge conflicts.

Done , I have merge the latest main branch to patch-1.

andrewhavck commented:
This has been included in our latest sync, thanks for contributing!
